src/BottomSheet/BottomSheet.js

```diff
@@ -0,0 +1,147 @@
+import React, {Component, PropTypes} from 'react';
+import ReactDOM from 'react-dom';
+import SlidingSheet from '../internal/SlidingSheet';
+
+class BottomSheet extends Component {
+  static propTypes = {
+    /**
+     * The contents of the `BottomSheet`
+     */
+    children: PropTypes.node,
+    /**
+     * The CSS class name of the root element.
+     */
+    className: PropTypes.string,
+    /**
+     * The CSS class name of the container element.
+     */
+    containerClassName: PropTypes.string,
+    /**
+     * Override the inline-styles of the container element.
+     */
+    containerStyle: PropTypes.object,
+    /**
+     * If true, the `BottomSheet` will be modal.
+     */
+    modal: PropTypes.bool,
+    /**
+     * Callback function fired when the `BottomSheet` is requested to be closed. If
+     * not provided, the `BottomSheet` will persist unless closed through the `open`
+     * prop.
+     *
+     * @param {string} reason The reason for the close request. Possible values are
+     * 'clickaway' (on overlay clicks), 'escape' (on escape key press), and 'swipe'
+     * (when swiping down).
+     */
+    onRequestClose: PropTypes.func,
+    /**
+     * If true, the `BottomSheet` is opened.
+     */
+    open: PropTypes.bool.isRequired,
+    /**
+     * The CSS class name to add to the `Overlay` component that is rendered behind the `BottomSheet`.
+     */
+    overlayClassName: PropTypes.string,
+    /**
+     * Override the inline-styles of the `Overlay` component that is rendered behind the `BottomSheet`.
+     */
+    overlayStyle: PropTypes.object,
+    /**
+     * Override the inline-styles of the root element.
+     */
+    style: PropTypes.object,
+    /**
+     * The width of the `BottomSheet` in pixels, if it should be inset. If not set, the sheet will be
+     * full-width.
+     */
+    width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+
+  };
+
+  static defaultProps = {
+    modal: false,
+    open: false,
+    width: '100%',
+  };
+
+  static contextTypes = {
+    muiTheme: PropTypes.object.isRequired,
+  };
+
+  componentWillMount() {
+    this.setState({
+      height: null,
+    });
+  }
+
+  componentDidMount() {
+    this.updateHeight();
+    window.addEventListener('resize', this.updateHeight);
```

Comment:
What about using this pattern https://github.com/callemall/material-ui/blob/master/src/Popover/Popover.js#L355?